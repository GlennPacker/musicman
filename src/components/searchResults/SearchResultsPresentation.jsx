import React from 'react';
import PropTypes from 'prop-types';
import { Container, Paper, Table, TableBody, TableCell, TableContainer, TableHead, TableRow, Typography } from '@material-ui/core';

const SearchResultsPresentation = ({searchResults}) => {
  return (
    <Container>
      { searchResults.length ? 
        <TableContainer component={Paper}>
          <Table>
            <TableHead>
              <TableRow>
                <TableCell>Artist</TableCell>
                <TableCell>Album</TableCell>
                <TableCell>Song</TableCell>
              </TableRow>
            </TableHead>
            <TableBody>
              {
                searchResults.map((row, index) => (
                  <TableRow key={ index }>
                    <TableCell component="th" scope="row">{ row.artist }</TableCell>
                    <TableCell>{ row.album }</TableCell>
                    <TableCell>{ row.song }</TableCell>
                  </TableRow>
                ))
              }
            </TableBody>
          </Table>
        </TableContainer>  : 
        <Typography>
            No items matching search terms
        </Typography>
      }
    </Container>
  );
};

SearchResultsPresentation.propTypes = {
  searchResults: PropTypes.array
};
 
export default SearchResultsPresentation;