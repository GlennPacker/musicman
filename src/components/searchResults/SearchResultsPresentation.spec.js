import SearchResultsPresentation from './SearchResultsPresentation';
import React from 'react';
import render from 'react-test-renderer';
import { mount } from 'enzyme';

describe('SearchResultsPresentation', () => {
  test('should render with no resuls', () => {
    const component = render.create(
      <SearchResultsPresentation 
        searchResults={ [] }
      />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render  with results', () => {
    const results = [{
      artist: 'abc',
      album: 'def',
      song: 'ghi'
    }];

    const component = render.create(
      <SearchResultsPresentation 
        searchResults={ results }
      />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('should be able to submit a valid search', () => {
    const wrapper = mount(
      <SearchResultsPresentation 
        searchResults={ [] } 
      />
    );

    expect(
      wrapper
        .text()
        .toLowerCase()
        .includes('no items matching search terms')
    ).toBe(true);
  });
});