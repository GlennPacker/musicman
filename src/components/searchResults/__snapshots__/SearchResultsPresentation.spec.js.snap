// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultsPresentation should render  with results 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg"
>
  <div
    className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <table
      className="MuiTable-root"
      role={null}
    >
      <thead
        className="MuiTableHead-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Artist
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Album
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Song
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
            role="cell"
            scope="row"
          >
            abc
          </th>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            def
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            ghi
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`SearchResultsPresentation should render with no resuls 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg"
>
  <p
    className="MuiTypography-root MuiTypography-body1"
  >
    No items matching search terms
  </p>
</div>
`;
